src/gennylib/include/gennylib/v1/PoolFactory.hpp

```diff
@@ -66,6 +67,8 @@ class PoolFactory {
      */
     void setOption(OptionType type, const std::string& option, std::string value);
 
+    void setEncryptionContext(std::shared_ptr<EncryptionContext> encryption);
```

Comment:
I decided to use shared pointer here because it's PoolManager that actually needs to maintain the EncryptionContext for all instances of this client pool name, and so that makes PoolManager the real "owner".